prow/gerrit/adapter/adapter.go

```diff
@@ -692,6 +697,28 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 	return nil
 }
 
+// lastRevisionWithCodeChange tries to find a revision with code change since last update.
+// There is no point repeating tests if no code change happened since last update.
+func lastRevisionWithCodeChange(change client.ChangeInfo, lastUpdate time.Time) (client.RevisionInfo, error) {
+	var revisionsWithCodeChange []client.RevisionInfo
+	for _, revision := range change.Revisions {
+		if revision.Kind != gerrit.NoCodeChange {
+			revisionsWithCodeChange = append(revisionsWithCodeChange, revision)
+		}
+	}
+	sort.Slice(revisionsWithCodeChange, func(i, j int) bool {
```

Comment:
Nit: Add comment that this sorts in reverse chronological order.

Reply:
Done